src/user/bans.js

```diff
@@ -63,33 +61,47 @@ module.exports = function (User) {
 		], callback);
 	};
 
-	User.isBanned = function (uid, callback) {
+	User.getBannedAndExpired = function (uid, callback) {
 		if (parseInt(uid, 10) <= 0) {
 			return setImmediate(callback, null, false);
 		}
-		async.waterfall([
-			async.apply(User.getUserFields, uid, ['banned', 'banned:expire']),
-			function (userData, next) {
-				var banned = userData && userData.banned;
-				if (!banned) {
-					return next(null, banned);
-				}
+		User.getUserFields(uid, ['banned', 'banned:expire'], function (err, userData) {
+			if (err) {
+				return callback(err);
+			}
+			callback(null, User.calcBanExpiredFromUserData(userData));
+		});
+	};
 
-				// If they are banned, see if the ban has expired
-				var stillBanned = !userData['banned:expire'] || Date.now() < userData['banned:expire'];
+	User.calcBanExpiredFromUserData = function (userData) {
+		return {
+			banned: !!userData.banned,
+			'banned:expire': userData['banned:expire'],
+			banExpired: userData['banned:expire'] <= Date.now() && userData['banned:expire'] !== 0,
+		};
+	};
 
-				if (stillBanned) {
-					return next(null, true);
-				}
-				async.parallel([
-					async.apply(db.sortedSetRemove.bind(db), 'users:banned:expire', uid),
-					async.apply(db.sortedSetRemove.bind(db), 'users:banned', uid),
-					async.apply(User.setUserFields, uid, { banned: 0, 'banned:expire': 0 }),
-				], function (err) {
-					next(err, false);
+	User.unbanIfBanExpired = function (uid, callback) {
```

Comment:
Personal preference here, but given these method names, perhaps we could change the behaviour of `bans.js` to not add methods to `User`, but instead, to create a new object `bans`, and then add methods to that with shorter names.

`User.ban.unbanIfExpired`? Might not be worth the hassle though :)

Reply:
sure, but is it ok to break all plugins that might be using `User.ban/unban`? or keep them with deprecation messages?

Reply:
We'll likely just keep this as-is for now, and clean up `bans.js` in a new PR